docs/content/en/docs/Advanced/bundles.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Bundles"
+linkTitle: "Bundles"
+weight: 5
+description: >
+    Bundles are a powerful feature of Kairos that allow you to customize and configure your operating system. This section explains how to use and build custom bundles.
+
+---
+
+Bundles are a powerful feature of Kairos that allow you to customize and configure your operating system, as well as your Kubernetes cluster. Whether you want to add custom logic, install additional packages, or make any other changes to your system, bundles make it easy to apply these changes after installation or before bootstrapping a node.
+
+Bundles are container images containing only files (and not full OS) that can be used to install new software or extend the cloud-init syntax. You can find community-supported bundles in the [community-bundles](https://github.com/kairos-io/community-bundles) repository.
+
+## Consuming Bundles
+
+To use a bundle in your Kairos configuration, you will need to specify the type of bundle and the target image in your cloud-config file.
+
+To apply a bundle before Kubernetes starts, you can include it in your config like this:
+
+```yaml
+#cloud-config
+
+bundles:
+- targets:
+  - run://<image>
+```
+
+Replace `<image>` with the URL or path to the bundle image. The prefix (e.g. `run://`) indicates the type of bundle being used.
+
+To install a bundle after installation instead (for those bundles that explicitly supports that), use the following:
+
+```yaml
+#cloud-config
+install:
+ bundles:
+ - targets:
+   - run://<image>
+```
+
+One of the benefits of using bundles is that they can also extend the cloud-config keywords available during deployment. This means that by adding bundles to your configuration file, you can add new blocks of configuration options and customize your system even further.
```

Comment:
We called it "installation" right above, let's be consistent:

```suggestion
One of the benefits of using bundles is that they can also extend the cloud-config keywords available during installation. This means that by adding bundles to your configuration file, you can add new blocks of configuration options and customize your system even further.
```

Reply:
if a bundle is installed after installation (as described above), then and `#cloud-config` it provides, will be ignored right? Maybe we should clarify that here.

Reply:
I didn't wanted to make it sound complex here as it's mostly an implication of how bundles deliver business logic. for instance, the bundles currently in community-bundle runs also before deployment of k3s (that is at the first boot), but let's iterate over this. I don't want to make this PR already too much big